src/ONV.cpp

```diff
@@ -0,0 +1,230 @@
+#include <bitset>
+#include "ONV.hpp"
+
+
+namespace GQCG {
+
+
+/*
+ *  PRIVATE METHODS
+ */
+
+/**
+ *  Extracts the positions of the set bits from the representation and places them in an array
+ */
+void ONV::update(){
+    size_t l = this->unsigned_representation;
+    int i = 0;
+    while(l != 0){
+        this->occupation_indexes(i) = __builtin_ctzl(l);
+        i++;
+        l ^= (l & -l);
+    }
+    if(i != this->N){
+        throw std::invalid_argument("current representation and electron count are not compatible");
+    }
+}
+
+
+/*
+ *  CONSTRUCTORS
+ */
+
+/**
+ *  Constructor from a @param K orbitals, N electrons and a representation for the ONV
+ */
+ONV::ONV(size_t K, size_t N, size_t representation): K(K), N(N), unsigned_representation(representation){
+    occupation_indexes = VectorXs::Zero(N);
+    update();  // throws error if the representation and N are not compatible
+}
+
+
+
+/*
+ *  OPERATORS
+ */
+
+/**
+ *  Overloading of operator<< for a GQCG::ONV to be used with streams
+ */
+std::ostream& operator<<(std::ostream& os, const GQCG::ONV& onv) {
+    const char* begin = reinterpret_cast<const char*>(&onv.unsigned_representation);
```

Comment:
this code is pretty ugly, could we use `boost::dynamic_bitset<>` to make the code cleaner?

Reply:
if you want to keep this code, explain it thoroughly